### DISCUSSION TITLE
Verification email not received

### DISCUSSION THREAD

**Discussion by syedfaiqyazdani** [documenso/documenso#1421] [General]
I am trying to create an account on documenso but it does not send verification email on my email can someone help me out ?

**Comment by indiedev84** [comment]
I use the self host version. I have the same problem. I thought it was related to the smtp server settings, but the letters with the link to reset the password come. And the link to confirm the registration is not. Resending does not solve the problem.

**Comment by 3011felix** [comment]
For me it's also not working. There is also nothing in the smtp logs.

**Comment by IZIme07** [comment]
the same

**Comment by ralfdigitalagency** [comment]
Same here. No solution?

**Comment by Decode1415** [comment]
Same here - using Docker compose.

**Comment by 4r7if3x** [comment]
A quick workaround would be opening the Postgres console:

```shell
psql -U $DB_USER -d $DB_NAME
```

And then executing:

```sql
UPDATE "User" SET "emailVerified" = NOW() WHERE "email" = 'YourEmail';
```

To manually bypass the email verification step.

**Reply by kamasuPaul** [reply]
I tried all the other solutions, but all seemed not to work. This is what worked for me

**Comment by flowq-C** [comment]
No responses regarding an issue that's been unresolved since November. Is it even worth pursuing this, if I can't even get past the registration stage.

**Comment by QuinnMatthews** [comment]
Spent a long time debugging this. Finally found the solution in an issue that I can't seem to locate anymore, but you need to set the `NEXT_PRIVATE_INTERNAL_WEBAPP_URL` env variable to `http://<name of container>:<port>` eg `http://documenso:3000`

**Reply by 4r7if3x** [reply]
I'm not sure how relevant this is, but I tried it anyway since it seemed harmless. However, it had no effect during my testing.

**Reply by Muntasir2001** [reply]
This worked for me.

No idea why or how as it makes no sense to me but oh well ....

**Reply by marcuscmy** [reply]
This worked for me as well.

Spent 2 days troubleshooting but at least its working now...

For those who are troubleshooting, when the request is successfully sent, there is a pop-up on the bottom right telling you the Confirmation email has been sent. There are no error dialogs otherwise.

**Reply by masteramyx** [reply]
Wow this was it. Thanks!

**Reply by mp0lski** [reply]
That did it for me too, thanks a lot!

**Comment by 4r7if3x** [comment]
Here are the logs in case they help:

```
at async /app/apps/web/.next/server/chunks/9783.js:1:17053
at async K (/app/node_modules/next/dist/compiled/next-server/pages-api.runtime.prod.js:20:16853)
at async U.render (/app/node_modules/next/dist/compiled/next-server/pages-api.runtime.prod.js:20:17492)
at async NextNodeServer.runApi (/app/node_modules/next/dist/server/next-server.js:600:9)
at async NextNodeServer.handleCatchallRenderRequest (/app/node_modules/next/dist/server/next-server.js:269:37)
at async NextNodeServer.handleRequestImpl (/app/node_modules/next/dist/server/base-server.js:816:17)
at async invokeRender (/app/node_modules/next/dist/server/lib/router-server.js:174:21)
at async handleRequest (/app/node_modules/next/dist/server/lib/router-server.js:353:24)

Submitting job to endpoint: https://contracts.example.com/api/jobs/send.signup.confirmation.email/cm97rx0qm000go55cv8rajjdl
Submitting job to endpoint: https://contracts.example.com/api/jobs/send.signup.confirmation.email/cm97rwydr0000o55c1thnp4ct

[JOBS]: Triggering job send.signup.confirmation.email with payload { email: 'REDACTED_EMAIL' }

Error: queryA EBADNAME localhost:2500
    at QueryReqWrap.onresolve [as oncomplete] (node:internal/dns/callback_resolver:45:19)
    at QueryReqWrap.callbackTrampoline (node:internal/async_hooks:128:17) {
  errno: undefined,
  code: 'EDNS',
  syscall: 'queryA',
  hostname: 'localhost:2500',
  command: 'CONN'
}

[JOBS]: Job send.signup.confirmation.email failed Error: Failed to send the confirmation email
    at u (/app/apps/web/.next/server/chunks/9783.js:1:34900)
    at async Object.handler (/app/apps/web/.next/server/chunks/9783.js:1:21019)

at runNextTicks (node:internal/process/task_queues:60:5)
at process.processImmediate (node:internal/timers:447:9)
at process.callbackTrampoline (node:internal/async_hooks:128:17)
```

And here are my email configs:

```ini
NEXT_PRIVATE_SMTP_TRANSPORT="smtp-auth"
NEXT_PRIVATE_SMTP_HOST="smtp.resend.com"
NEXT_PRIVATE_SMTP_PORT=465
NEXT_PRIVATE_SMTP_SECURE=false
NEXT_PRIVATE_SMTP_USERNAME="resend"
NEXT_PRIVATE_SMTP_PASSWORD="REDACTED_RESEND_API_KEY"
NEXT_PRIVATE_SMTP_FROM_ADDRESS="no-reply@mailer.example.com"
NEXT_PRIVATE_SMTP_FROM_NAME="Contracts"
```

I also tried using secure mode with port `587`, but it didn’t work. Additionally, configuring `resend` as the transport with the `NEXT_PRIVATE_RESEND_API_KEY` environment variable didn’t work either.

---

P.S. I also noticed that if you manually bypass the signup confirmation step, some types of emails are sent successfully, while others still fail.

**Reply by 4r7if3x** [reply]
Perhaps @andris9 from @nodemailer could shed some light on what’s going wrong, @ElTimuro @Mythie.

**Reply by QuinnMatthews** [reply]
Do you mind if I ask, what does the rest of your docker-compose/.env file (redacted as necessary) look like?

**Reply by 4r7if3x** [reply]
Well, I’m using the [template](https://github.com/coollabsio/coolify/blob/main/templates/compose/documenso.yaml) provided by [Coolify](https://github.com/coollabsio/coolify), and I haven’t modified any of its defaults or auto-generated values, except for the ones above.

**Reply by Patil5913** [reply]
i have been using the template provided by dokploy and have been facing the same issue from last one week. this is the docker compose file. i have tried both 587 as well as 465 but its not working anyway.

version: "3.8"
services:
  postgres:
    image: postgres:16
    volumes:
      - documenso-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_USER=documenso
      - POSTGRES_PASSWORD=password
      - POSTGRES_DB=documenso
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U documenso"]
      interval: 10s
      timeout: 5s
      retries: 5
      start_period: 10s

  documenso:
    image: documenso/documenso:v1.5.6-rc.2
    command: sh -c "npm install sharp && npm start" 
    depends_on:
      postgres:
        condition: service_healthy
    environment:
      - PORT=${DOCUMENSO_PORT}
      - NEXTAUTH_URL=http://${DOCUMENSO_HOST}
      - NEXTAUTH_SECRET=${NEXTAUTH_SECRET}
      - NEXT_PRIVATE_ENCRYPTION_KEY=${NEXT_PRIVATE_ENCRYPTION_KEY}
      - NEXT_PRIVATE_ENCRYPTION_SECONDARY_KEY=${NEXT_PRIVATE_ENCRYPTION_SECONDARY_KEY}
      - NEXT_PUBLIC_WEBAPP_URL=http://${DOCUMENSO_HOST}
      - NEXT_PRIVATE_DATABASE_URL=my db url
      - NEXT_PRIVATE_DIRECT_DATABASE_URL=my db url
      - NEXT_PUBLIC_UPLOAD_TRANSPORT=database
      - NEXT_PRIVATE_SMTP_TRANSPORT=smtp
      - NEXT_PRIVATE_SMTP_HOST=smtp.hostinger.com
      - NEXT_PRIVATE_SMTP_PORT=587
      - NEXT_PRIVATE_SMTP_SECURE=false
      - NEXT_PRIVATE_SMTP_USERNAME=test@vrugle.com
      - NEXT_PRIVATE_SMTP_PASSWORD=mypassword
      - NEXT_PRIVATE_SMTP_FROM_EMAIL=test@vrugle.com
      - NEXT_PRIVATE_SMTP_FROM_NAME="Notifications"
      - NEXT_PRIVATE_SIGNING_LOCAL_FILE_PATH=/opt/documenso/cert.p12
    ports:
      - ${DOCUMENSO_PORT}
    volumes:
      - /opt/documenso/cert.p12:/opt/documenso/cert.p12

volumes:
  documenso-data:

**Reply by samuelbohl** [reply]
I am exactly in the same situation as @4r7if3x

**Comment by Novapixel1010** [comment]
Is anyone here still having this issue? Have you checked  #1763 that is what worked for me

**Reply by Dvalin21** [reply]
I've updated to the new version and I constantly get a Auth Plain error, auth failed.  I noticed that you have the IGNORE TLS set as true, wont that ignore any certs including starttls?

logs
'''
Error: Invalid login: 535 5.7.8 Error: authentication failed: (reason unavailable)
documenso-app  |     at SMTPConnection._formatError (/app/node_modules/nodemailer/lib/smtp-connection/index.js:809:19)
documenso-app  |     at SMTPConnection._actionAUTHComplete (/app/node_modules/nodemailer/lib/smtp-connection/index.js:1588:34)
documenso-app  |     at SMTPConnection.<anonymous> (/app/node_modules/nodemailer/lib/smtp-connection/index.js:556:26)
documenso-app  |     at SMTPConnection._processResponse (/app/node_modules/nodemailer/lib/smtp-connection/index.js:993:20)
documenso-app  |     at SMTPConnection._onData (/app/node_modules/nodemailer/lib/smtp-connection/index.js:774:14)
documenso-app  |     at SMTPConnection._onSocketData (/app/node_modules/nodemailer/lib/smtp-connection/index.js:195:44)
documenso-app  |     at TLSSocket.emit (node:events:518:28)
documenso-app  |     at TLSSocket.emit (node:domain:489:12)
documenso-app  |     at addChunk (node:internal/streams/readable:561:12)
documenso-app  |     at readableAddChunkPushByteMode (node:internal/streams/readable:512:3) {
documenso-app  |   code: 'EAUTH',
documenso-app  |   response: '535 5.7.8 Error: authentication failed: (reason unavailable)',
documenso-app  |   responseCode: 535,
documenso-app  |   command: 'AUTH PLAIN'
'''
Yes my password is correct.

**Comment by Dvalin21** [comment]
SO I figured it out for myself.    1. If your using port 587 then - NEXT_PRIVATE_SMTP_SECURE needs to be "false" If 465, then "true"  I don't remember this being made clear in the documentation. Also, the # OPTIONAL: Leave blank to allow users to signup through /signup page.
NEXT_PUBLIC_DISABLE_SIGNUP="" should just be a tick or check box to disable once the initial account its logged into. Instead you have edit the env and shut down and restart. There should be a invite function if you need to someone to join that you can initiate through the first account and/or allowing the customer or person who your sending something to sign to be able to create an account before or after they sign.

**Comment by Dvalin21** [comment]
UPDATE:    After the update to the documentation for Documenso, the SMTP env has been updated and made simpler.
Verified. All you need now is 

# [[SMTP]]
# OPTIONAL: Defines the transport to use for sending emails. Available options: smtp-auth (default)
NEXT_PRIVATE_SMTP_HOST="fqn"
NEXT_PRIVATE_SMTP_PORT="port"
NEXT_PRIVATE_SMTP_USERNAME="username"
NEXT_PRIVATE_SMTP_PASSWORD="password"
NEXT_PRIVATE_SMTP_FROM_NAME="Account name"
NEXT_PRIVATE_SMTP_FROM_ADDRESS="email address"

**Comment by catalinpit** [comment]
Seems like this is resolved now, so it can be closed.
